src/hoc/ErrorBoundary.tsx

```diff
@@ -1,96 +1,34 @@
-import React, { ReactElement } from 'react';
-import history from '../history';
+import React from 'react';
 import { IMainStrings, IState } from '../model';
 import { connect } from 'react-redux';
 import localStrings from '../selector/localize';
-import { bindActionCreators } from 'redux';
-import * as actions from '../store';
-import {
-  Typography,
-  Accordion,
-  AccordionSummary,
-  AccordionDetails,
-  styled,
-} from '@mui/material';
-import ExpandMoreIcon from '@mui/icons-material/ExpandMore';
-import {
-  logError,
-  Severity,
-  forceLogin,
-  localUserKey,
-  LocalKey,
-} from '../utils';
+import { logError, Severity } from '../utils';
 import { withBucket } from './withBucket';
-import Memory from '@orbit/memory';
-import { PriButton } from '../control';
-
-const ModalDiv = styled('div')(() => ({
-  position: 'fixed' /* Stay in place */,
-  zIndex: 1 /* Sit on top */,
-  paddingTop: '100px' /* Location of the box */,
-  left: 0,
-  top: 0,
-  width: '100%' /* Full width */,
-  height: '100%' /* Full height */,
-  overflow: 'auto' /* Enable scroll if needed */,
-  backgroundColor: 'rgba(0, 0, 0, 0.4)' /* Black w/ opacity */,
-}));
-
-const ModalContentDiv = styled('div')(() => ({
-  backgroundColor: '#fefefe',
-  margin: 'auto',
-  padding: '20px',
-  border: '1px solid #888',
-  width: '80%',
-  height: '40%',
-  textAlign: 'center',
-  display: 'flex',
-  flexDirection: 'column',
-  '& #detail': {
-    textAlign: 'left',
-    fontSize: 'small',
-  },
-}));
-
-const ModalActionsDiv = styled('div')(() => ({
-  display: 'flex',
-  flexDirection: 'row',
-  justifyContent: 'center',
-}));
+import { ModalMessage } from '../components/ErrorPage';
 
 interface IStateProps {
   t: IMainStrings;
   orbitStatus: number | undefined;
   orbitMessage: string;
   orbitDetails?: string;
-  orbitRetry: number;
-}
-
-interface IDispatchProps {
-  resetOrbitError: typeof actions.resetOrbitError;
 }
 
-interface IProps extends IStateProps, IDispatchProps {
+interface IProps extends IStateProps {
   errorReporter: any;
-  memory: Memory;
-  resetRequests: () => Promise<void>;
   children: JSX.Element;
 }
 
 const initState = {
   errCount: 0,
   error: '',
   details: '',
-  expanded: false,
   view: '',
 };
 
 export class ErrorBoundary extends React.Component<IProps, typeof initState> {
   constructor(props: IProps) {
     super(props);
-    this.continue = this.continue.bind(this);
-    this.logout = this.logout.bind(this);
-    this.change = this.change.bind(this);
+    this.handleReset = this.handleReset.bind(this);
```

Comment:
why did you have to do this and how did you know you had to do it?

Reply:
I did it because it is part of the pattern used by class components. The methods have to be bound such that the instance variable is passed as the first parameter. I knew I needed to do it because with version 1, I had all class components. (Previously it was the preferred way to write react.) This all changed with the advent of hooks. But it turns out the "componentDidCatch" isn't something that there is a hook for so we still have this one component that we implement as a class component.